src/server/common/helpers/auth/federated-oidc.js

```diff
@@ -0,0 +1,239 @@
+import * as Hoek from '@hapi/hoek'
+import * as openid from 'openid-client'
+import Joi from 'joi'
+import Boom from '@hapi/boom'
+
+import { getAADCredentials } from '~/src/server/common/helpers/auth/cognito.js'
+import { config } from '~/src/config/config.js'
+import { createLogger } from '~/src/server/common/helpers/logging/logger.js'
+import { sessionNames } from '~/src/server/common/constants/session-names.js'
+import { asExternalUrl } from '~/src/server/common/helpers/url/url-helpers.js'
+
+const logger = createLogger()
+const callbackPath = '/auth/callback'
+
+export const federatedOidc = {
+  name: 'federatedOidc',
+  register: function (server) {
+    server.auth.scheme('federated-oidc', scheme)
+    server.auth.strategy('azure-oidc', 'federated-oidc', {
+      discoveryUri: config.get('oidcWellKnownConfigurationUrl'),
+      redirectUri: config.get('appBaseUrl') + callbackPath,
+      clientId: config.get('azureClientId'),
+      scope: `api://${config.get('azureClientId')}/cdp.user openid profile email offline_access user.read`,
+      tokenProvider: getAADCredentials
+    })
+  }
+}
+
+const scheme = function (server, options) {
```

Comment:
why is this function annoymous? This won't help you when debugging. Suggestion. Change it to 
`function scheme(server, options) {`